c++ training/SteamProject/Videogame/Steam.cpp

```diff
@@ -0,0 +1,434 @@
+	std::cin >> TemporalUserChoice;
+
+	return ValidateInputAndReturnValidatedInput(TemporalUserChoice, "TELL ME A VALID OPTION: ", 1, CategoryContainer.GetCurrentNumberOfCategories()) - 1;
+}
+//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+bool FSteam::AskChooseCategory()
+{
+	system("cls");
+	std::string AskChooseCategoryMessage = "Would you like to choose a category for your new game? \n1 - Yes\n2 - No\nTELL ME : ";
+	std::cout << AskChooseCategoryMessage;
+
+	enum class EQuestionCategoryCommand { Yes = 1, No };
+	EQuestionCategoryCommand UserChoice;
+
+	float TemporalUserChoice;
+	std::cin >> TemporalUserChoice;
+
+	UserChoice = (EQuestionCategoryCommand)ValidateInputAndReturnValidatedInput(TemporalUserChoice, AskChooseCategoryMessage, 1, 2);
+
+	switch (UserChoice)
+	{
+	case EQuestionCategoryCommand::Yes:
+		return true;
+
+	case EQuestionCategoryCommand::No:
+		return false;
+
+	default:
+		std::cout << "AskChooseCategoryMenu Error, please contact support";
+		return false;
+	}
+}
+//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+bool FSteam::IsInteger(float Float) const
+{
+	return (Float - floor(Float)) == 0.f;
+}
+//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+bool FSteam::IsInRange(float Value, int LowerBound, int UpperBound) const
+{
+	return Value >= LowerBound && Value <= UpperBound;
+}
+//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+int FSteam::ValidateInputAndReturnValidatedInput(float Input, std::string Message, int LowerBound, int UpperBound)
+{
+	while (!std::cin.good() || !IsInteger(Input) || !IsInRange(Input, LowerBound, UpperBound))
+	{
+		ResetConsoleInputScreen();
+
+		std::cout << Message;
+		std::cin >> Input;
+	}
+	return (int)Input;
+}
+//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+std::string FSteam::ValidateInputAndReturnValidatedInput(std::string StringInput, std::string Message)
+{
+	while (!std::cin.good() || StringInput == "")
+	{
+		ResetConsoleInputScreen();
+
+		std::cout << Message;
+		std::cin >> StringInput;
+	}
+	return StringInput;
+}
+//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+int FSteam::GetValidDay(float TemporalDay, int Month, int Year)
```

Comment:
This one could also be const after making your validation methods constant. This is an example of the issues with having even just one method not follow const-correctness: you're forced to have more non-const methods.